boxes/generators/platonic.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+# Copyright (C) 2013-2016 Florian Festi
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from boxes import *
+from boxes.edges import FingerJointBase, FingerJointEdge
+
+from math import sin, pi
+
+class UnevenJointBase(FingerJointBase):
+    def calcFingers(self, length, bedBolts):
+        space, finger = self.settings.space, self.settings.finger
+        fingers = int((length - (self.settings.surroundingspaces - 1) * space) //
```

Comment:
Using surroundingspaces doesn't really make sense for this kind of edge. It allows the FingerJointEdge to squeeze one finger in less length than one finger and two spaces. Unless you have a the leftovers on a different height this doesn't make any sense.